src/Core/ExceptionWrapper.php

```diff
@@ -35,12 +34,41 @@ class ExceptionWrapper
      * Adds response body to the exception message.
      *
      * @param BadResponseException $ex
-     * @return GraphException containing HTTP response from Graph API
-     * 
+     * @return BadResponseException containing HTTP response from Graph API
      */
     public static function wrapGuzzleBadResponseException(BadResponseException $ex)
     {
-        $errMsg = "Received {$ex->getResponse()->getStatusCode()} for call to {$ex->getRequest()->getUri()}\nAPI response: {$ex->getResponse()->getBody()->getContents()}";
-        return new GraphException($errMsg);
+        $response = $ex->getResponse();
+
+        // Safety check for Guzzle < 7.0
+        if (!$response) {
+            return $ex;
+        }
+
+        /** @see \GuzzleHttp\Exception\RequestException::create() */
+        if (preg_match('/^(.+: `.+ .+` resulted in a `.+ .+` response)($|:\n)/U', $ex->getMessage(), $match)) {
+            $message = $match[1];
+
+            $body = $response->getBody();
+            $summary = $body->getContents();
+            if ($body->isSeekable()) {
+                $body->rewind();
+            }
+
+            if ($summary !== '') {
+                $message .= ":\n{$summary}\n";
+
+                //return new $ex($message, $ex->getRequest(), $ex->getResponse(), $ex, $ex->getHandlerContext());
+                // Better: modify internal message inside original exception object (preserves the stack trace)
+                (new class() extends \Exception {
+                    public function overwriteProtectedMessage(\Exception $ex, $message)
```

Comment:
Curious why this needs to be in an anonymous class

Reply:
Internal implementation detail? Actually there are other ways to do `$ex->message = $message;` (which we can't do directly because the property is `protected`):
```php
(function () use ($message) { $this->message = $message; })->call($ex);
```
or:
```php
$rp = new \ReflectionProperty(\Exception::class, 'message');
$rp->setAccessible(true);
$rp->setValue($ex, $message);
```
but I find them more "risky"... WDYT?

Reply:
Makes sense considering the property is `protected`.